tinyinstinstrumentation.cpp

```diff
@@ -68,7 +74,17 @@ RunResult TinyInstInstrumentation::Run(int argc, char **argv, uint32_t init_time
       WARN("Target function not reached, retrying with a clean process\n");
       instrumentation->Kill();
       cur_iteration = 0;
-      status = instrumentation->Run(argc, argv, init_timeout);
+      if (attach_mode) {
+        Sleep(init_timeout);
+        processID = FindProcessId(process_name);

```

Comment:
There should be a check here if FindProcessId failed. Instead of calling `Sleep` first and then `FindProcessId`, maybe you can do something like
```
while(1) {
  processID = FindProcessId(process_name);
  if(processID) break;
  WARN('Could not find target process, retrying after timeout');
  Sleep(init_timeout);
}